infrastructure/builder/src/generate/generators/node-version.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('fs');
+const path = require('path');
+const {readFile, modifyFile, modifyJSON} = require('../util');
+
+/**
+ * Update the node version to match everywhere, treating that in `package.json`
+ * as authoritative.
+ */
+exports.tasks = [{
+  title: 'Node Version',
+  provides: ['target-node-version'],
+  run: async (requirements, utils) => {
+    const nodeVersion = JSON.parse(await readFile('package.json')).engines.node;
+    if (!nodeVersion || !nodeVersion.match(/[0-9.]+/)) {
+      throw new Error(`invalid node version ${nodeVersion} in package.json`);
+    }
+    utils.step({title: `Setting node version ${nodeVersion}`});
+
+    utils.status({message: '.taskcluster.yml'});
+    await modifyFile('.taskcluster.yml',
+      contents => contents.replace(
+        /^( *node: ')[0-9.]+(')$/m,
+        `$1${nodeVersion}$2`));
```

Comment:
Is the plan to keep this as a regular expression search and replace?  Parsing a yaml document as a string can be a little dangerous.

Reply:
I'm not totally happy with it either, but I don't know of tools that can reliably re-write a file including all its comments, etc.  If you know of something, let me know!  At any rate, this will produce diffs that people will see, so if something goes wrong then we'll notice.